azure-kusto-data/src/request_options.rs

```diff
@@ -0,0 +1,549 @@
+    query_results_apply_getschema: Option<bool>,
+    /// If positive, controls the maximum age of the cached query results the service is allowed to return
+    query_results_cache_max_age: Option<KustoDuration>,
+    /// If set, enables per-shard query cache.
+    query_results_cache_per_shard: Option<bool>,
+    /// Hint for Kusto as to how many records to send in each update (takes effect only if OptionResultsProgressiveEnabled is set)
+    query_results_progressive_row_count: Option<i64>,
+    /// Hint for Kusto as to how often to send progress frames (takes effect only if OptionResultsProgressiveEnabled is set)
+    query_results_progressive_update_period: Option<i32>,
+    ///  Enables limiting query results to this number of records.
+    query_take_max_records: Option<i64>,
+    /// Controls query consistency
+    queryconsistency: Option<QueryConsistency>,
+    /// Request application name to be used in the reporting (e.g. show queries).
+    request_app_name: Option<String>,
+    /// If specified, blocks access to tables for which row_level_security policy is enabled
+    request_block_row_level_security: Option<bool>,
+    /// If specified, indicates that the request can't call-out to a user-provided service.
+    request_callout_disabled: Option<bool>,
+    /// Arbitrary text that the author of the request wants to include as the request description.
+    request_description: Option<String>,
+    /// If specified, indicates that the request can't invoke code in the ExternalTable.
+    request_external_table_disabled: Option<bool>,
+    /// If specified, indicates that the service should not impersonate the caller's identity.
+    request_impersonation_disabled: Option<bool>,
+    /// If specified, indicates that the request can't write anything.
+    request_readonly: Option<bool>,
+    ///  If specified, indicates that the request can't access remote databases and clusters.
+    request_remote_entities_disabled: Option<bool>,
+    /// If specified, indicates that the request can't invoke code in the sandbox.
+    request_sandboxed_execution_disabled: Option<bool>,
+    /// Request user to be used in the reporting (e.g. show queries).
+    request_user: Option<String>,
+    /// If set, enables the progressive query stream
+    results_progressive_enabled: Option<bool>,
+    /// Overrides the default request timeout.
+    servertimeout: Option<KustoDuration>,
+    /// Overrides the default maximum number of records a query is allowed to return to the caller (truncation).
+    truncationmaxrecords: Option<i64>,
+    /// Overrides the default maximum data size a query is allowed to return to the caller (truncation).
+    truncationmaxsize: Option<i64>,
+    /// Validates user's permissions to perform the query and doesn't run the query itself.
+    validate_permissions: Option<bool>,
+}
+
+#[derive(Default)]
+pub struct RequestOptionsBuilder(RequestOptions);
```

Comment:
Can we cut down on the duplication? maybe a builder macro?

Reply:
A macro would be great, there even is one in the azure core crate. However it forces us to use less idiomatic names. Here i put more weight on having "with_*" style names. Now that you mention it, its even inconsistent between the request builders and here.

Then again, i was unsure how much we save when the function names and property names differ, which i feel we should keep. The raw property names are sometimes a bit harder to work with.

Happy to hear your thoughts. 

Related issue in azure sdks:
https://github.com/Azure/azure-sdk-for-rust/issues/292

Reply:
Maybe we want to take a step back and use something that is meant specifically for builders, and not just setters. lIke https://github.com/colin-kiegel/rust-derive-builder , which will let us add prefixes and save on some boilerplate.
But it is another dependency, so it's a question

Reply:
hi @AsafMah - had a look into the derive crate you mentioned.. looks very promising. As you mentioned this would be another dependency, but I believe the transient dependncies that would get added are quite minimal, since that crate only consumes very popular crates...

I guess what I am saying is that I would trust your judgement if we want to introduce that crate :).

Reply:
Take a look at https://github.com/Azure/azure-kusto-rust/pull/18
I experimented with adding it